source/views/directory/list.tsx

```diff
@@ -16,29 +16,69 @@ import {NoticeView, LoadingView} from '@frogpond/notice'
 import {formatResults} from './helpers'
 import {useFetch} from 'react-async'
 import {List, Avatar} from 'react-native-paper'
-import type {DirectoryItem, SearchResults} from './types'
+import type {
+	DirectorySearchTypeEnum,
+	DirectoryItem,
+	SearchResults,
+} from './types'
 import Icon from 'react-native-vector-icons/Ionicons'
-import {useNavigation} from '@react-navigation/native'
-import {NativeStackNavigationOptions} from '@react-navigation/native-stack'
-import {ChangeTextEvent} from '../../navigation/types'
+import {RouteProp, useNavigation, useRoute} from '@react-navigation/native'
+import {
+	NativeStackNavigationOptions,
+	NativeStackNavigationProp,
+} from '@react-navigation/native-stack'
+import {ChangeTextEvent, RootStackParamList} from '../../navigation/types'
+
+const getDirectoryUrl = (query: string, type: DirectorySearchTypeEnum) => {
+	let baseUrl = 'https://www.stolaf.edu/directory/search?format=json'
+
+	switch (type) {
+		case 'Query':
+			return `${baseUrl}&query=${query.trim()}`
+		case 'Department': {
```

Comment:
Add case Major?

Reply:
I don't really have a way to test that state or that endpoint -- do we want to go down supporting that if we're not using it?